### DISCUSSION TITLE
Custom migrations can collide with official migrations

### DISCUSSION THREAD

**Discussion by infomiho** [directus/directus#14826] [Feature Requests]
### Describe the Bug

When doing some new feature, we used a version for migration `20220308A` which collided with one of the internal migrations [20220308A-add-bookmark-icon-and-color.ts](https://github.com/directus/directus/blob/main/api/src/database/migrations/20220308A-add-bookmark-icon-and-color.ts)

The culprit is here https://github.com/directus/directus/blob/main/api/src/database/migrations/run.ts#L28 - there is not differentiation between custom and internal migrations, which probably should be addressed. The user shouldn't track the list of internal migrations writing his own.

Maybe prefix the custom migration versions? (Not sure how to make this backwards compatible) Or maybe emphasise in the docs to name the custom migrations `20220308customA-create-table.js` 

### To Reproduce

Create a custom migration file `20220308A-test-migration.js`

### Errors Shown

> Error: Migration keys collide! Please ensure that every migration uses a unique key.

### What version of Directus are you using?

9.13.0

**Comment by rijkvanzanten** [comment]
The key before the first hyphen is an arbitrary string, so that could be anything. A note in the docs sounds like the most helpful thing here 🙂 Wanna open a PR for that? https://github.com/directus/docs

cc @erondpowell

**Reply by infomiho** [reply]
It's an arbitrary string - but look at the recommended way of writing it in the docs: https://docs.directus.io/extensions/migrations/#file-name

I'll submit a PR sometime tomorrow 👍

**Reply by erondpowell** [reply]
@infomiho
Did we ever do a PR for this? I couldn't spot it.
If we haven't yet, I can do it!

**Reply by rijkvanzanten** [reply]
Don't think so! https://github.com/directus/docs/pulls?q=is%3Apr+author%3Ainfomiho+

**Reply by infomiho** [reply]
Hey guys, I haven't had the time to write it.

@erondpowell if you would do it, that would be amazing. Thanks!

**Reply by piotr-cz** [reply]
Note about suffix on custom migration identifier in docs would really help here.
Now I have to write a migration to rename custom migrations as identifiers collide in new Directus version

**Reply by piotr-cz** [reply]
For anyone who is tempted to write migration to migrate current custom migrations - it's now worth it.
Just rename custom migration files and entries in database. 

I went with the `20201202A_custom-my-custom-migration.js` file structure.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
